provisioning/ci/azure_keyVault.tf

```diff
@@ -0,0 +1,66 @@
+resource "azurerm_key_vault" "go_cloud_encrypt" {
+  name                = "${var.name_prefix}-go-cloud-encrypt"
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  resource_group_name = azurerm_resource_group.go_cloud_encrypt.name
+  location            = azurerm_resource_group.go_cloud_encrypt.location
+  sku_name            = "standard"
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = azuread_service_principal.go_cloud_encrypt.id
+
+    key_permissions = [
+      "Get",
+      "List",
+      "Update",
+      "Create",
+      "Import",
+      "Delete",
+      "Recover",
+      "Backup",
+      "Restore",
+      "Purge",
+      "Encrypt",
+      "Decrypt",
+    ]
+  }
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azuread_group.developers.object_id
+
+    key_permissions = [
+      "Get",
+      "List",
+      "Update",
+      "Create",
+      "Import",
+      "Delete",
+      "Recover",
+      "Backup",
+      "Restore",
+      "Purge",
+      "Encrypt",
+      "Decrypt",
+    ]
+  }
+}
+
+resource "azurerm_key_vault_key" "go_cloud_encrypt" {
+  name         = "${var.name_prefix}-go-cloud-encrypt"
+  key_vault_id = azurerm_key_vault.go_cloud_encrypt.id
+  key_type     = "RSA"
+  key_size     = 2048
```

Comment:
Why do we use asymetric encryption when library supports AES256 symetric encryption?

Reply:
Is any of these symmetric?

![image](https://github.com/user-attachments/assets/bc383784-2d08-4601-b769-7510731498cc)

Reply:
none, I am just maybe wrong, but I see that our native encryptor uses symetric encryption and decryption, but the KeyVault uses asymetric one. Maybe there is some logic behind it, I can ask Odin for more info why it is like that or maybe that's azure specific part.
Simply, asymetric encyption/decryption is more expensive

Reply:
https://keboolaglobal.slack.com/archives/C055HSMKX51/p1732020699339909